[PLT-7344] Revamp - Certification/Testing Page

amnambiar opened the pull request:
https://input-output.atlassian.net/browse/PLT-7344
https://input-output.atlassian.net/browse/PLT-7052

BE: [PR #90](https://github.com/input-output-hk/dapps-certification/pull/90) `origin/feat/l1-certificate` (also contains profile field modifications)


@coderabbitai: ignore


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
### Summary by CodeRabbit

Release Notes:

- New Feature: Introduced `Session`, `Landing`, `Certification`, and `TestingHistory` routes for enhanced navigation.
- New Feature: Added an `icon` prop to the `Button` component for improved customization.
- Refactor: Simplified HTML head section in `public/index.html` for better performance.
- Refactor: Updated `onRequestError` function in `src/api/onRequest.ts` to handle unauthorized requests more effectively.
- Refactor: Replaced `Loader` component with `CircularProgress` as the fallback for suspense in `App.tsx`.
- Refactor: Modified `CreateCertificate` component to use updated keys from `state.auth` object.
- Style: Enhanced visual appearance of primary button style and log view layout.
- Test: Prepared test cases for `LogsView` component (currently commented out).
- Chore: Updated color definitions and added new properties in `tailwind.config.js`.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

amnambiar committed:
PLT-7342 revamped layout

Commit:
Replaced @mui/styles with tss-react to fix React 18 compatibility

Commit:
Added tailwind css, removed reset css and using mui css baseline

Commit:
Added landing page

Commit:
Added fetch tiers, mocked fetch price and registation form validations

Commit:
Updated wallet integration

amnambiar committed:
PLT-7439 Refined useLocalStorage hook to prevent/handle setting of incorrect values into keys

amnambiar committed:
Merge branch 'PLT-7439' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
Merge branch 'PLT-7439' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

Commit:
Updated landing integration

amnambiar committed:
WIP

amnambiar committed:
Merge branch 'PLT-7109' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
PLT-7344 Added Auditor Run-Test Form and some API Integrations

amnambiar committed:
PLT-7344 Styled html table with tailwind for timeline view

amnambiar committed:
Merge branch 'PLT-7047' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
PLT-7344 Auditor test form reset logic + other fixes

amnambiar committed:
PLT-7344 Added tooltip to Input

amnambiar committed:
PLT-7344 Input bg color update

amnambiar committed:
PLT-7344 Subject required + RepoAccessStatus added

Commit:
Updated redux slices

Commit:
Finished integrations

Commit:
Added wallet changes integration

Commit:
Fixed integration

Commit:
Updated navigation

Commit:
Updated pages and component from master

Commit:
Added header

Commit:
Merge branch 'master' into PLT-7342

Commit:
Merged PLT-7455 changes (Switch to stake-address & Yoroi support)

amnambiar committed:
Merge branch 'PLT-7342' into PLT-7344
- Updated tailwind color classnames across + resolved merge conflicts

amnambiar committed:
PLT-7342 Removed Certification and Added Auditor Report Upload with icon

amnambiar committed:
Merge branch 'PLT-7342' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

Commit:
Updated testing history page

Commit:
Added node-sass

Commit:
Fixed sort icon

Commit:
Resolved PRs comments

amnambiar committed:
PLT-7342 To handle the case where subscriptions array is empty (case when pending subscription is deleted through the API)

amnambiar committed:
PLT-7342 Renamed balance variable and other text

amnambiar committed:
Merge branch 'PLT-7342' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
PLT-7344 Integrated PR#9 inand Certification

Commit:
Resolved comments

Commit:
Fixed hook call

Commit:
Fixed session local storage access

Commit:
Fixed scrolling and report upload page

amnambiar committed:
Merge branch 'PLT-7342' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
PLT-7344 Further updates to full report view

amnambiar committed:
Merge branch 'PLT-7353' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

Commit:
Fixed missing comma

amnambiar committed:
PLT-7344 Some minor fixes

amnambiar committed:
PLT-7344 More updates to Full Report View

amnambiar committed:
PLT-7344 Further updates to Full Report View

amnambiar committed:
PLT-7344 Full Report Success/Failure details

amnambiar committed:
PLT-7344 Fixes across

amnambiar committed:
PLT-7344 Fix responsive form alignments + wrap tooltip within Input/TextArea

amnambiar committed:
PLT-7344 Verify Repository access

amnambiar committed:
PLT-7734 Floating label in textarea when prefilled values

amnambiar committed:
PLT-7344 State retaining

amnambiar committed:
Merge branch 'PLT-7344' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

RSoulatIOHK commented:
## Full report
### Scrolling
When looking at the full report, there are 2 horizontal scroll bars. Can we have at most 1 or even better 0 and line wrap the text? 

There is a vertical scroll bar but it's kind of hidden on the far right and you need to use the right horizontal scroll bar to find it. Can we have it, on the right, at all times? I don't think people use the scroll bar to actually navigate but it will display that there is more information to see.

https://github.com/input-output-hk/dapps-certification-web/assets/117812549/50a6cfee-80c2-4db5-aa5e-8dd1563d4c8f

### Display more information
It is unclear on this screen that you can click on the properties to show more information
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/f1bff8c5-21b4-46b0-8748-da0fc53394ac)
Maybe just an arrow pointing downwards, with still the possibility to click anywhere you want to open the details.

### "Purchase a certificate"
The "Purchase a Certificate (1 ADA)" should not be here for failed runs.
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/08c5d322-c1ff-420a-9df4-1ec02f3490d4)

## Testing 

### Logs
The log opens all over everything.
Can we have it open in between the timeline and the PBT results?

### "Blank" page at the end of testing
At the end of testing, the timeline is removed and no testing results is displayed to the user:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/dc3ba731-ae41-4138-819b-3facfd5a907e)

Could we have instead this kind of nice visualization, without the possibility to open the properties to see more detail results?
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/f1bff8c5-21b4-46b0-8748-da0fc53394ac)

### Testing time still here from a precedent run
The testing time is still displayed from a previous run:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/5b7de913-b78e-4fb8-a4ae-370a1d9a764f)

### Test again
Pressing test again gives you this screen:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/4d182e03-b11d-47c3-82b5-6dda7f00b3e7)

Not too sure what's going on, I think a new timeline has been fitted in the middle.
The testing information on the left is empty.
The previous "Full report" and "Download report" from the previous run should not be here anymore.

amnambiar committed:
Merge branch 'PLT-7047' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
PLT-7344 Modifications to Full Report table + Show timeline always in TimelineView

amnambiar committed:
PLT-7344 Minor fixes across UI in Form

amnambiar committed:
PLT-7344 Removed ability for commit hash url in repo

amnambiar committed:
PLT-7344 Fixed vertical scroll and sticky header in layout

amnambiar committed:
PLT-7344 Fixed Button component in Certification

amnambiar committed:
PLT-7344 Fixed state retention

amnambiar committed:
Merge branch 'PLT-7047' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344
- Few other tailwind class/style fixes

amnambiar committed:
PLT-7344 Fix to horizontal scroll in Full Report Table

amnambiar committed:
PLT-7734 Show CreateCertificate only if unit test or other tasks are all success; dont show if any failure

amnambiar committed:
PLT-7734 Positioning of Logs

amnambiar commented:
@RSoulatIOHK , Fixes are in place to re-review.

**Full Report View**
> Scrolling

FIXED

> Display more information - Maybe just an arrow pointing downwards, with still the possibility to click anywhere you want to open the details.

DONE ; added up/down clickable arrows to see the details, retaining click on row 

> The "Purchase a Certificate (1 ADA)" should not be here for failed runs.

FIXED ; This button will not be present if there any failures in unit tests or other tasks

**Testing View**
> Logs - Can we have it open in between the timeline and the PBT results?

DONE

> "Blank" page at the end of testing

FIXED ; the same table while the run in-progress will be shown

> Testing time still here from a precedent run

FIXED ; all other issues related to state retention are as well fixed, [PLT-7245](https://input-output.atlassian.net/browse/PLT-7425) as well

> Test again - bug

FIXED ; all states in _New test_ and _Test again_ are fixed

**Others**
- Fixed Button styles

**Open**
- Github Repo Access needs to be thoroughly tested; also [PLT-7458](https://input-output.atlassian.net/browse/PLT-7458)
- While Testing is running, the % progress in properties within the table are fine. But once it is finished, the % of progress of the last property is not correctly updated.
- Property Based Testing card needs to integrated correctly
- Integrate PR #8 
- Cannot see all tasks within Full Report table, due to [PLT-7614](https://input-output.atlassian.net/browse/PLT-7614)

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,158 @@
-                />
-              )}
+    // Prefill form values
+    useEffect(() => {
+        if (lsFormData && lsUuid && !runId) {
+            const data: IAuditorRunTestFormFields = JSON.parse(
+                JSON.stringify(lsFormData)
+            );
+            const { repoURL, commit: commitHash, version, name, subject } = data;
+            if (repoURL && commitHash && version && name && version && subject) {
+                const [, , , username, repoName] = data.repoURL.split("/");
+                setFormData(data);
+                onCertificationFormSubmit({
+                    runId: lsUuid as string,
+                    commitHash: commitHash,
+                    repo: username + "/" + repoName,
+                });
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [lsFormData]);
```
check for !runId ensures it is triggered only when there is no active run in place. Ignoring

amnambiar replied:
ignoring

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,158 @@
-    } catch (e) {
-      handleErrorScenario();
-      console.error('Failed:', e);
+    const resetStates = (ended: boolean = false) => {
+        setRunEnded(ended)
+        setRunId("")
+        setDisableForm(false)
+        setClickedFormSubmit(false)
     }
```
ignoring

amnambiar commented on tailwind.config.js:
```diff
@@ -1,31 +1,76 @@
       backgroundImage: {
         landing: 'url(assets/images/landing-background.png)',
       },
+      borderRadius: {
+        3: "3px",
+      },
+      padding: {
+        20: "20px",
+        22: "22px"
+      },
+      margin: {
+        5: "5px",
+        10: "10px",
+        20: "20px"
+      },
```
ignoring

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,158 @@
+                            className="button text-sm min-w-[150px]"
+                            startIcon={<LeaderboardIcon />}
+                        >New test</Button>
+                    </div>
+                )}
+            </div>
+            <div
+                id="certificationWrapper"
+                className="content-area-box shadow-lg bg-white px-5 xs:px-7 xs:py-4 flex flex-col tab:flex-row tab:px-5"
+            >
+                <div className="sm:w-full tab:w-1/2 px-0 mb-6 tab:px-22 tab:mb-0">
+                    <CertificationForm
+                        loadingRunId={()=> { setClickedFormSubmit(true) }}
+                        onSubmit={onCertificationFormSubmit}
+                        disable={disableForm}
+                        testAgain={testAgain}
+                        clearForm={clearForm}
+                        initialData={formData}
+                        onError={certificationFormError}
+                    />
+                </div>
+                <div className="sm:w-full tab:w-1/2 px-22 min-h-[150px] tab:px-22 tab:mb-0">
+                    {runId ? (
+                        <TimelineView
+                            uuid={runId}
+                            repo={repo}
+                            commitHash={commitHash}
+                            runEnded={onRunEnd}
+                            onAbort={onTestRunAbort}
+                            triggerFormReset={onTestRunAbort}
+                        />
+                    ) : (
+                        <div className="w-full text-center text-xl text-neutral-300 font-medium pt-48">
+                            {clickedFormSubmit ? <Loader /> : <span>Fill the testing form</span>}
+                        </div>
+                    )}
+                </div>
             </div>
-          </Form>
         </div>
```
ignoring

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,158 @@
+    const [clickedFormSubmit, setClickedFormSubmit] = useState(false);
+    const [formData, setFormData] = useState<IAuditorRunTestFormFields | null>(null);
+
+    const onCertificationFormSubmit = (data: {
+        runId: string;
+        commitHash: string;
+        repo: string;
+    }) => {
+        setRunId(data.runId);
+        setRunCommitHash(data.commitHash);
+        setRepo(data.repo)
+        setDisableForm(true);
     };
```
ignoring

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,158 @@
-        console.error('Failed:', e);
-      }
+    // const { features: subscribedFeatures } = useAppSelector((state) => state.auth);
+    const [lsFormData] = useLocalStorage(LocalStorageKeys.certificationFormData, "");
+    const [lsUuid] = useLocalStorage(LocalStorageKeys.certificationUuid, "");
```
ignoring

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,316 @@
+  const dispatch = useAppDispatch();
+  const confirm = useConfirm();
+
+  const [runStatus, setRunStatus] = useState("");
+  const [runState, setRunState] = useState("");
+  const [refetchMin, setRefetchMin] = useState(5);
+  const [coverageFile, setCoverageFile] = useState("");
+  const [fetchRunStatus, setFetchRunStatus] = useState(false);
+  const [apiFetching, setApiFetching] = useState(false);
+  const [timelineConfig, setTimelineConfig] = useState(TIMELINE_CONFIG);
+  const [resultData, setResultData] = useState<any>({});
+  const [unitTestSuccess, setUnitTestSuccess] = useState(true); // assuming unit tests will pass
+  const [plannedTestingTasks, setPlannedTestingTasks] = useState<PlanObj[]>([]);
```
ignoring

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,316 @@
+  const [unitTestSuccess, setUnitTestSuccess] = useState(true); // assuming unit tests will pass
+  const [plannedTestingTasks, setPlannedTestingTasks] = useState<PlanObj[]>([]);
+
+  const abortTestRun = () => {
+    confirm({ title: "", description: "Are sure you want to abort this run!" })
+      .then(async () => {
+        await dispatch(
+          deleteTestHistoryData({ url: "/run/" + uuid + "?delete=true" })
+        );
+        onAbort();
+      })
+      .catch(() => {});
+  };
```
ignoring for now

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,316 @@
+  };
+
+  const triggerFetchRunStatus = async () => {
+    let config = timelineConfig;
+    try {
+      const res = await fetchData.get("/run/" + uuid);
+      const status = res.data.status;
+      const state = res.data.hasOwnProperty("state") ? res.data.state : "";
+      setRunStatus(status);
+      setRunState(state);
+      setFetchRunStatus(state === "running" || state === "passed");
+      config = processTimeLineConfig(config, state, status, res);
+      handleTestingTimelineDetails(status, state, res);
+      if (status === "finished") {
+        const isArrayResult = Array.isArray(res.data.result);
+        const resultJson = isArrayResult
+          ? res.data.result[0]
+          : res.data.result;
+        if (isArrayResult) {
+          setCoverageFile(res.data.result[1]);
+        }
+        setResultData(resultJson)
+        runEnded(true)
+        const unitTestResult = processFinishedJson(resultJson);
+        setUnitTestSuccess(unitTestResult);
+        // navigate to result page
+        clearPersistentStates();
+        // navigate("/report/" + uuid, { state: { repoUrl: githubLink, certifiable: true } });
+      }
+      if (state === "failed" || status === "finished") {
+        // setSubmitting(false);
+        clearPersistentStates();
+        setApiFetching(false);
+      }
+      setTimelineConfig(config);
+    } catch (e) {
+      // handleErrorScenario();
+      console.error("Failed:", e);
+      triggerFormReset();
+    }
```
ignoring for now

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+  onError,
+  loadingRunId
+}) => {
+  const form: any = useForm({
+    schema: auditorRunTestFormSchema,
+    mode: "all",
+  });
```
ignoring

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+  const [searchParams, setSearchParams] = useSearchParams();
+  const githubAccessCode = searchParams.get("code");
+
+  const connectToGithub = async () => {
+    try {
+      // fetch CLIENT_ID from api
+      const clientId = (await fetchData.get("/github/client-id"))
+        .data as string;
+      window.location.assign(
+        `https://github.com/login/oauth/authorize?client_id=${clientId}&scope=repo`
+      );
+    } catch (errorObj: any) {
+      let errorMsg = "Unable to connect to the repository. ";
+      if (errorObj?.response?.data) {
+        errorMsg +=
+          errorObj.response.statusText + " - " + errorObj.response.data + ". ";
+      }
+      errorMsg += "Please recheck and try again.";
+      setShowError(errorMsg);
+      const timeout = setTimeout(() => {
+        clearTimeout(timeout);
+        setShowError("");
+      }, 5000);
+    }
+  };
```
ok

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+    setSubmitting(true);
+    loadingRunId && loadingRunId(true)
+    const triggerAPI = async () => {
+      try {
+        const checkout = commitHash || commit;
+        const profileResponse = await fetchData.put("/profile/current", {
+          ...profile,
+          dapp: {
+            owner: username,
+            repo: repoName,
+            name: name,
+            version: version,
+            subject: subject
+          },
+        });
+        if (profileResponse.data.dapp.owner) {
+          const response = await postData.post("/run", checkout);
+          if (response.data) {
+            // store data into LS
+            setLsUuid(response.data)
+            setLsFormData({
+              ...formData,
+              commit: checkout,
+              repoURL: `https://github.com/${username}/${repoName}`,
+            })
+            // Emit uuid, checkout
+            onSubmit({
+              runId: response.data,
+              commitHash: checkout,
+              repo: username + "/" + repoName
+            });
+          }
+        }
+      } catch (e) {
+        handleError(e);
+      }
+      setSubmitting(false);
+    };
+    triggerAPI();
+  };
```
ignoring for now

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+  // Lists all repo and have to manually click 'Grant' for the repo we need.
+  // If that is not chosen, it doesn't ask again for permission
+  useEffect(() => {
+    // to extract ?code= from the app base url redirected after Github connect
+    if (githubAccessCode) {
+      (async () => {
+        await dispatch(getUserAccessToken({ code: githubAccessCode }));
+        searchParams.delete("code");
+        setSearchParams(searchParams);
+
+        const formData = form.getValues();
+        const [, , , username, repoName] = formData.repoURL.split("/");
+
+        const timeout = setTimeout(async () => {
+          clearTimeout(timeout);
+          await dispatch(verifyRepoAccess({ owner: username, repo: repoName }));
+          localStorage.removeItem(LocalStorageKeys.accessToken);
+        }, 0);
+      })();
+    }
+    // the enclosed snippet is to be triggered only once right when the component is rendered to check if the url contains code (to validate if it is a redirect from github)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
ignoring for now

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [repoUrlChanges]);
+
+  useEffect(() => {
+    showConfirmConnection && confirmConnectModal();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [showConfirmConnection]);
```
ignoring

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/AuditorRunTestForm.tsx:
```diff
@@ -0,0 +1,326 @@
+          <Input
+            label="Commit Hash"
+            type="text"
+            id="commit"
+            required={true}
+            disabled={submitting}
+            {...form.register("commit")}
+          />
+
+          <Input
+            label="DApp Name"
+            type="text"
+            id="name"
+            required={true}
+            disabled={submitting}
+            {...form.register("name")}
+          />
+
+          <Input
+            label="DApp Version"
+            type="text"
+            id="version"
+            required={true}
+            disabled={submitting}
+            {...form.register("version")}
+          />
+
+          <TextArea
+            placeholder="DApp Subject"
+            maxRows={2}
+            required={true}
+            disabled={submitting}
+            {...form.register("subject")}
+          />
+        </div>
+      </Form>
+
+      {showError ? <Toast message={showError} /> : null}
+    </div>
+  );
```
ignoring for now

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,316 @@
+    }
+  }, [runStatus, runState]);
+
+  useDelayedApi(
+    async () => {
+      setFetchRunStatus(false); // to clear timeout until api response
+      triggerFetchRunStatus();
+    },
+    refetchMin * TIMEOFFSET, // delay in milliseconds
+    fetchRunStatus // set to false to stop polling
+  );
```
ignoring for now

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,316 @@
+    };
+  }, []);
+
+  if (!runStatus) {
+    return <Loader />
+  }
```
ignoring

amnambiar commented on src/components/RepoAccessStatus/RepoAccessStatus.tsx:
```diff
@@ -0,0 +1,26 @@
+import classNames from "classnames";
+import { ACCESS_STATUS } from "./repoAccessStatus.config";
+
+const RepoAccessStatus: React.FC<{
+  status: string;
+  statusText?: string;
+}> = ({ status, statusText }) => {
```
ok; set to ""

amnambiar commented on src/components/RepoAccessStatus/RepoAccessStatus.tsx:
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable jsx-a11y/alt-text */
+import classNames from "classnames";
+import { ACCESS_STATUS } from "./repoAccessStatus.config";
+
+const RepoAccessStatus: React.FC<{
+  status: string;
+  statusText?: string;
+}> = ({ status, statusText }) => {
+  if (!status) {
+    return null
+  }
+  
+  const { color, className, ...imgProps } = ACCESS_STATUS[status];
+  return (
+    <label className="info flex">
+      <img style={{marginTop: "-4px"}} className={classNames("h-5 w-5 bg-white", className)} {...imgProps} />
+      {statusText ? (
+        <span style={{ color: color }} className="ml-2">
+          {statusText}
+        </span>
+      ) : null}
+    </label>
+  );
+};
+
+export default RepoAccessStatus;
```
ignoring for now

amnambiar commented on src/pages/certification/components/DownloadResult/DownloadResult.tsx:
```diff
@@ -0,0 +1,20 @@
+
+const DownloadResult = (resultData: any) => {
+
+  const handleDownloadResultData = (resultData: any) => {
+    exportObjectToJsonFile(resultData, "Testing Report.json");
+  };
```
ignoring for now

amnambiar commented on src/pages/certification/components/DownloadResult/DownloadResult.tsx:
```diff
@@ -0,0 +1,20 @@
+import { exportObjectToJsonFile } from "utils/utils";
+import DownloadIcon from '@mui/icons-material/Download';
+
+const DownloadResult = (resultData: any) => {
+
+  const handleDownloadResultData = (resultData: any) => {
+    exportObjectToJsonFile(resultData, "Testing Report.json");
+  };
+
+  return <Button
+      className="button"
+      onClick={(_) => handleDownloadResultData(resultData)}
+      variant="contained" size="medium"
+      startIcon={<DownloadIcon />}
+    >
+      Download Report
+    </Button>
+}
+export default DownloadResult;
```
ignoring for now

amnambiar commented on src/components/RepoAccessStatus/repoAccessStatus.config.ts:
```diff
@@ -0,0 +1,23 @@
+export const ACCESS_STATUS: { [index: string]: any } = {
+    verifying: {
+      className: "image anim-rotate",
+      "data-testid": "verifying",
+      src: "images/running.svg",
+      alt: "verifying",
+      color: "#DBAB0A",
+    },
+    accessible: {
+      className: "image",
+      "data-testid": "accessible",
+      src: "images/passed.svg",
+      alt: "accessible",
+      color: "#009168",
+    },
+    notAccessible: {
+      className: "image",
+      "data-testid": "notAccessible",
+      src: "images/failed.svg",
+      alt: "not accessible",
+      color: "#FF493D",
+    },
+  };
```
ignoring for now

amnambiar commented on src/components/ProgressCard/CircularProgressBar.tsx:
```diff
@@ -0,0 +1,40 @@
+import classNames from "classnames";
+
+const CircularProgressBar = ({
+  progress = 0,
+  color = "stroke-pink-400",
+}: {
+  progress: number;
+  color?: string;
+}) => {
+  return (
+    <div className="relative">
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="-1 -1 34 34"
+        className="progress-bar"
+      >
+        <circle
+          cx="16"
+          cy="16"
+          r="15.9155"
+          className={classNames("fill-none stroke-2 opacity-25", color)}
+        ></circle>
+
+        <circle
+          cx="16"
+          cy="16"
+          r="15.9155"
+          className={classNames("progress-bar-track fill-none stroke-2", color)}
+          style={{ strokeDashoffset: 100 - progress }}
+        ></circle>
+      </svg>
+
+      <span className="absolute top-1/2 left-0 progress-text  w-full text-center text-2xl font-extrabold">
+        {progress}%
+      </span>
+    </div>
+  );
+};
+
+export default CircularProgressBar;
```
done

amnambiar commented on src/hooks/useLocalStorage.ts:
```diff
@@ -4,39 +4,97 @@ import { Dispatch, SetStateAction, useEffect, useState } from "react";
+        parsedValue = item;
+      }
+
+      if (typeof parsedValue === 'string' && initialValue === null) {
+        // if the initialValue of key is supposed to be null, and the stored value is a string, 
+        // then it is wrongly set and need to be cleared
+        return initialValue;
+      } else if (parsedValue === null) { 
+        // in case of possible null after JSON.parse
+        return initialValue;
+      } else {
+        return parsedValue;
+      }
```
Proceeded with this as per review in another PR; ignoring this comment

amnambiar commented on src/hooks/useLocalStorage.ts:
```diff
@@ -52,21 +110,26 @@ function useLocalStorage<T>(
       try {
-        const serializedValue = typeof newValue === 'string' ? newValue : JSON.stringify(newValue);
-        window.localStorage.setItem(key, serializedValue);
+        if (newValue !== "" || newValue !== null) {
+          const serializedValue = typeof newValue === "string" ? newValue : JSON.stringify(newValue);
+          window.localStorage.setItem(key, serializedValue);
+
+          // Save state
+          setStoredValue(newValue);
+
+          // We dispatch a custom event so every useLocalStorage hook are notified
+          window.dispatchEvent(new Event("local-storage"));
+        } else {
+          removeValue()
+        }
```
done

amnambiar commented on src/pages/certification/Certification.helper.tsx:
```diff
@@ -84,4 +80,17 @@ export interface Log {
+  return CertificationTasks.filter(item => item.type === type).map(item => item.key);
+}
+
+export const getCertificationTaskName = (key: string) => {
+  const obj: {key: string, label: string} | any = CertificationTasks.find(item => item.key === key)
+  return obj.label;
+};
```
ignoring

amnambiar commented on src/components/ProgressCard/ProgressCard.tsx:
```diff
@@ -0,0 +1,48 @@
+  displayText?: string;
+  color?: string;
+  tooltipText?: string
+}> = ({ color = "stroke-green", currentValue, totalValue, displayText, title, tooltipText }) => {
+  const [progress, setProgress] = useState(0)
+
+  useEffect(() => {
+    setProgress(Math.trunc((currentValue/totalValue) * 100))
+  }, [currentValue, totalValue])
```
Done

amnambiar commented on src/components/ProgressCard/ProgressCard.tsx:
```diff
@@ -0,0 +1,48 @@
+}> = ({ color = "stroke-green", currentValue, totalValue, displayText, title, tooltipText }) => {
+  const [progress, setProgress] = useState(0)
+
+  useEffect(() => {
+    setProgress(Math.trunc((currentValue/totalValue) * 100))
+  }, [currentValue, totalValue])
```
Done

amnambiar commented on src/components/ProgressCard/ProgressCard.tsx:
```diff
@@ -0,0 +1,48 @@
+    setProgress(Math.trunc((currentValue/totalValue) * 100))
+  }, [currentValue, totalValue])
+
+  const card = () => {
+    return (
+      <div className="flex max-w-full  content-between shadow shadow-lg bg-white rounded-sm flex-col p-3 sm:flex-row w-fit">
+        <div className="inline-flex flex-col justify-around">
+          <h5 className="text-neutral truncate max-w-full">
+            {title}
+          </h5>
+          {displayText ? 
+            <span className="inline-block font-medium text-neutral-900 text-sm">{displayText}</span> 
+            : 
+            <span className="inline-block font-semibold text-2xl">
+              {currentValue} / {totalValue}
+            </span>
+          }
+        </div>
+        <div className="mt-5 sm:mt-0 w-fit">
+          <CircularProgressBar progress={progress} color={color} />
+        </div>
+      </div>
+    );
+  }
```
done

amnambiar commented on src/components/Table/components/ColViz/ColViz.tsx:
```diff
@@ -57,34 +67,44 @@ const ColViz: FC<any> = ({ columns, updateColumnOptions }) => {
+          </ListItemIcon>
+          <ListItemText>{column.Header}</ListItemText>
+        </MenuItem>
+      ))}
```
done

amnambiar commented on src/components/Button/Button.tsx:
```diff
@@ -25,12 +26,13 @@ const Button: FC<IButtonProps> = ({
   size = "medium",
   showLoader = false,
   displayStyle = "primary",
+  icon = null,
   onClick,
   ...props
 }) => {
   return (
     <button
-      className={`btn ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
+      className={`px-3 py-2 btn bg-secondary hover:bg-blue rounded-3 ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
       type={type}
       onClick={onClick}
       disabled={showLoader || disabled}
```
ignoring

amnambiar commented on src/components/Button/Button.tsx:
```diff
@@ -25,12 +26,13 @@ const Button: FC<IButtonProps> = ({
   return (
     <button
-      className={`btn ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
+      className={`px-3 py-2 btn bg-secondary hover:bg-blue rounded-3 ${className} ${isLoading ? "is-loading" : ""} ${"btn-" + displayStyle} ${"btn-" + size}`}
```
ignoring

amnambiar commented on src/hooks/useLogs.ts:
```diff
@@ -125,7 +120,7 @@ export const useLogs = (
                 dispatch(setBuildInfo())
             }
         } catch(e) {
-            handleErrorScenario();
+            handleErrorScenario && handleErrorScenario();
             console.log(e);
         } finally{
             setFetchingLogs(false)
```
ignoring

amnambiar commented on src/components/Table/components/ColViz/ColViz.tsx:
```diff
@@ -17,17 +27,17 @@ const ColViz: FC<any> = ({ columns, updateColumnOptions }) => {
     });
     atLeastOneItemChecked(updatedColumns);
     // setColumnDetails(updatedColumns);
   };
```
done

amnambiar commented on src/api/onRequest.ts:
```diff
@@ -17,7 +17,10 @@ export function onRequest(config: AxiosRequestConfig) {
 }
 
 export function onRequestError(error: AxiosError) {
-  // Do something with request error
+  if (localStorage.getItem(LocalStorageKeys.isLoggedIn) !== "true") {
+    localStorage.clear();
+    window.location.reload();
+  }
   return Promise.reject(error);
 }
```
ignoring

amnambiar commented on tailwind.config.js:
```diff
@@ -1,34 +1,76 @@
   },
-  plugins: [],
+  plugins: []
 };
```
done

amnambiar commented on src/pages/certification/certification-result/FullReportTable.tsx:
```diff
@@ -0,0 +1,69 @@
+  data,
+}) => {
+
+    const tableData = processData(data)
+
```
done

amnambiar commented on src/components/LogsView/LogsView.scss:
```diff
@@ -20,12 +42,14 @@
             float: left;
+            margin: 0;
         }
         .minimize-btn {
             cursor: pointer;
             font-size: xxx-large;
+            float: right;
         }
```
ignoring

amnambiar commented on src/pages/certification/certification-result/FullReportTable.tsx:
```diff
@@ -0,0 +1,69 @@
+
+    const tableData = processData(data)
+
+    const columns = useMemo(() => [
+      {
+        Header: "Property Name",
+        accessor: "label",
+      },
+      {
+        Header: "Status",
+        accessor: "status",
+        classes: "text-center w-[100px]",
+        disableSortBy: true,
+        Cell: (props: any) => {
+          if (props.row.original.status === 'success') {
+            return <img
+              className="image"
+              src="/images/passed.svg"
+              alt="success"
+            />
+          } else if (props.row.original.status === 'failure') {
+            return <img
+              className="image"
+              src="/images/failed.svg"
+              alt="failure"
+            />
+          } else { return null; }
+        }
+      }
+    ],
+      [])
```
done

amnambiar commented on src/pages/certification/certification-result/FullReportTable.tsx:
```diff
@@ -0,0 +1,69 @@
+        accessor: "status",
+        classes: "text-center w-[100px]",
+        disableSortBy: true,
+        Cell: (props: any) => {
+          if (props.row.original.status === 'success') {
+            return <img
+              className="image"
+              src="/images/passed.svg"
+              alt="success"
+            />
+          } else if (props.row.original.status === 'failure') {
+            return <img
+              className="image"
+              src="/images/failed.svg"
+              alt="failure"
+            />
+          } else { return null; }
+        }
```
ignoring

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,306 @@
+  const abortTestRun = () => {
+    confirm({ title: "", description: "Are sure you want to abort this run!" })
+      .then(async () => {
+        await dispatch(
+          deleteTestHistoryData({ url: "/run/" + uuid + "?delete=true" })
+        );
+        onAbort();
+      })
```
ignoring for now

amnambiar commented on src/components/LogsView/LogsView.tsx:
```diff
@@ -1,45 +1,58 @@
+
+
+    useEffect(() => {
+        bottomRef.current?.scrollIntoView({ behavior: 'smooth' }); // scroll to bottom 
+    }, [logs])
```
ignoring

amnambiar commented on src/components/LogsView/LogsView.tsx:
```diff
@@ -54,15 +67,15 @@ const InformationTable: FC<{logs: any, }> = ({ logs, }) => {
                                 if (attr?.fields?.hasOwnProperty('chunk-data')) {
                                     logData = attr['fields']['chunk-data']
                                 }
-                            } catch(e) {
+                            } catch (e) {
                                 // do nothing
                                 if (typeof item.Text == 'string' && item.Text.length) {
                                     logData = item.Text
                                 }
                             }
                             logData = !logData.length ? item.Text : logData
```
ignoring for now

amnambiar commented on src/pages/certification/components/FileCoverageContainer.tsx:
```diff
@@ -75,39 +76,43 @@ const FileCoverageContainer: React.FC<{
+            </div>
+        ) : null}
+    </>);
 };
```
ignoring for now

amnambiar commented on src/pages/certification/components/TimelineView/TimelineView.tsx:
```diff
@@ -0,0 +1,306 @@
+  };
+
+  const triggerFetchRunStatus = async () => {
+    let config = timelineConfig;
+    try {
+      const res = await fetchData.get("/run/" + uuid);
+      const status = res.data.status;
+      const state = res.data.hasOwnProperty("state") ? res.data.state : "";
+      setRunStatus(status);
+      setRunState(state);
+      setFetchRunStatus(state === "running" || state === "passed");
+      config = processTimeLineConfig(config, state, status, res);
+      handleTestingTimelineDetails(status, state, res);
+      if (status === "finished") {
+        const isArrayResult = Array.isArray(res.data.result);
+        const resultJson = isArrayResult
+          ? res.data.result[0]
+          : res.data.result;
+        if (isArrayResult) {
+          setCoverageFile(res.data.result[1]);
+        }
+        setResultData(resultJson)
+        runEnded(true)
+        const unitTestResult = processFinishedJson(resultJson);
+        setUnitTestSuccess(unitTestResult);
+        // navigate to result page
+        clearPersistentStates();
+        // navigate("/report/" + uuid, { state: { repoUrl: githubLink, certifiable: true } });
+      }
+      if (state === "failed" || status === "finished") {
+        // setSubmitting(false);
+        clearPersistentStates();
+        setApiFetching(false);
+      }
+      setTimelineConfig(config);
+    } catch (e) {
+      // handleErrorScenario();
+      console.error("Failed:", e);
+      triggerFormReset();
+    }
```
ignoring for now

amnambiar commented on src/pages/testingHistory/index.tsx:
```diff
@@ -233,10 +233,18 @@ const TestHistory = () => {
     window.open(url, "_blank");
   };
 
+  const formatRepoUrl = (repoUrl: string) => {
+    let pieces = repoUrl.split('github:')[1].split('/')
+    return (pieces[0] + "/" + pieces[1]);
+  }
+
   const columns = React.useMemo(() => [
     {
       Header: "Repo name",
       accessor: "repoUrl",
+      Cell: (props: any) => (
+        <span>{formatRepoUrl(props.row.original.repoUrl)}</span>
+      )
     },
```
ignoring for now

amnambiar commented on src/pages/certification/Certification.tsx:
```diff
@@ -1,343 +1,155 @@
+            }
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [lsFormData]);
```
ignoring for now

amnambiar commented on src/pages/certification/components/AuditorRunTestForm/auditorRunTestForm.schema.ts:
```diff
@@ -0,0 +1,76 @@
+export const COMMIT_EXIST_PATTERN = /\/commit\/[a-f0-9]{7,40}/;
+
+export const auditorRunTestFormSchema = yup.object().shape({
+  repoURL: yup
+    .string()
+    .required("This field is required")
+    .test({
+      name: "github-commit-url",
+      test: (value, context) => {
+        if (value) {
+          const [, , , username, repoName] = value!.split("/");
+          if (
+            !username ||
+            !repoName ||
+            !/^[a-zA-Z0-9-]+$/.test(username) ||
+            !/^[a-zA-Z0-9-]+$/.test(repoName)
+          ) {
+            return context.createError({
+              message: "Invalid GitHub repository URL",
+            });
+          }
+
+          const matches = value!.match(REPO_URL_PATTERN);
+          if (!matches) {
+            return context.createError({
+              message: "Invalid GitHub repository URL",
+            });
+          }
+
+          return true;
+        } else {
+          return true;
+        }
+      },
+    }),
```
done

amnambiar committed:
PLT-7344 AI Review comments done

amnambiar committed:
Merge branch 'PLT-6763' of https://github.com/input-output-hk/dapps-certification-web into PLT-7608

amnambiar committed:
PLT-7608 Renamed component

RSoulatIOHK commented:
Still have the double scrolling bars and the vertical one is far away to see.
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/c109465d-d755-45d3-894b-e61850b34d75)

When the repo doesn't exist, I get this Uncaught error
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/da72e996-55bc-43a0-95d7-5cf096d62b5b)

Tool tip and check for repo accessibility are over one another:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/d7ec2cac-fa89-42e2-9384-420acebab11a)

I know it's in the open issues but the progress % are not fine for most of the properties on some repo testing:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/b76a91a4-dd79-470f-92a8-125051c46747)
But is correct on the full report:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/07d4ab94-3cee-4e3a-87ab-28309ae72d72)

I still see the previous testing time:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/fa64824b-f31c-4505-9814-2604e1d3e9b2)

Also the Github repo check is a bit agressive. I haven't managed to type the repo myself without triggering the check, could we just have it when you're no longer in the input bar?

All the rest looks fixed :)

amnambiar committed:
PLT-7344 Fixed RepoAccess

amnambiar committed:
PLT-7344 Fix RepoAccess revalidate after Abort

amnambiar committed:
PLT-7344 Fix -clear run times on new run

amnambiar committed:
PLT-7344 Disable maxwidth in FullReport Table mui-Container to fix for responsive issue with multiple scrollbars

amnambiar commented:
@RSoulatIOHK , Thank you for the patience :bowing_woman: 

> Still have the double scrolling bars and the vertical one is far away to see.

Fixed for good now. Pardon :disappointed: 
Reason - mui-Container element had `maxWidth`defaulted to `'lg'`. I had changed it to `'xl'` instead of setting it `false`. Is set to `false` now.

> - When the repo doesn't exist, I get this Uncaught error
> - Tool tip and check for repo accessibility are over one another:

Fixed

> I still see the previous testing time:

Fixed :sweat_smile: 

>  Github repo check - could we just have it when you're no longer in the input bar

Triggering the check on blur breaks at multiple scenarios, hence retained the check on type itself but with a lesser timeout than the older version of tool. It is more quicker, doesn't let [PLT-7458 ](https://input-output.atlassian.net/browse/PLT-7458) happen (Test btn will be disabled until the repo is accessible).

> progress % issue

Raised [PLT-7646](https://input-output.atlassian.net/browse/PLT-7646) to track this

amnambiar committed:
PLT-7344 Force repo access revalidation on test again

amnambiar committed:
Merge branch 'PLT-7344' of https://github.com/input-output-hk/dapps-certification-web into PLT-7608

amnambiar committed:
PLT-7608 Prefill values within CertificationMetadata form; get profile coming into AppLayout to be used across the tool

amnambiar committed:
PLT-7608 Fixed button styles in metadata form

RSoulatIOHK commented:
I still have the scroll bars :( If I put the window on my 4k screen, i don't have them but that might be a strong requirement for our users 😄 
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/c3de965f-1932-411b-8fc2-f3c70d6e05e5)
(Let's put this one on the Tech debt if you want)

Rest is fine :) (apart from the tickets you have raised to track some of the bigger issues)

amnambiar committed:
PLT-7344 - PLT-7656 Fix to multiple horizontal scrollbars above 4k

amnambiar committed:
Merge branch 'PLT-7047' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar commented:
@RSoulatIOHK ,

> Multiple scrollbars above 4k 

I've added a fix for the multiple scrollbars. Hope it is resolved for you now. Anyhow I've raised a ticket [PLT-7656](https://input-output.atlassian.net/browse/PLT-7656). We can track it there.

Thanks!

RSoulatIOHK reviewed (approved):
Looks good!

amnambiar committed:
Merge branch 'master' of https://github.com/input-output-hk/dapps-certification-web into PLT-7344

amnambiar committed:
Merge branch 'PLT-7344' of https://github.com/input-output-hk/dapps-certification-web into PLT-7608

amnambiar committed:
PLT-7608 - Revert - prefill form fields with name,email,twitter,website from profile

amnambiar commented:
Rechecked 'Testing' the form, timeline view, full report, state retention and it looks fine.

RSoulatIOHK commented:
Apart from the issues in the timeline results when it ends (sometimes)

![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/367ce334-6718-413a-9e82-0e064199166c)

The rest looks good!

amnambiar committed:
PLT-7608 Fixed variable

amnambiar committed:
Merge pull request #20 from input-output-hk/PLT-7608

[PLT-7608] Migrate Auditor Sign Certificates
